.github/workflows/build-asset.yml

```diff
@@ -0,0 +1,116 @@
+name: Build and test asset
+
+on:
+  schedule:
+    - cron: '0 10 * * 1'
+  workflow_dispatch:
+    inputs:
+      versions:
+        description: 'Node.js version'
+        required: true
+        default: '[ "lts/*", "18" ]'
+  pull_request:
+    branches: [ master ]
+  push:
+    branches: [ master ]
+    tags: [ '*' ]
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ windows-latest, macos-latest, ubuntu-latest ]
+        version: ${{ fromJSON(inputs.versions || '[ "lts/*", "18" ]') }}
+    env:
+      NODEJS_VERSION: ${{ matrix.version }}
+      NEXE_ASSET: ${{ github.workspace }}/nexe-asset
+      NEXE_TMP_CACHE_PATH: ${{ github.workspace }}/nexe-tmp
+      NEXE_TMP: ${{ github.workspace }}/nexe-tmp/standard
+      CCACHE_COMPRESS: '1'
+      ErrorView: NormalView
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '${{ env.NODEJS_VERSION }}'
+      - name: Record Node version
+        run: |
+          echo "NODEJS_VERSION=$(node -p process.versions.node)" >> $env:GITHUB_ENV
+          echo "NODEJS_VERSION=$(node -p process.versions.node)" >> "${GITHUB_ENV}"
+      - run: npm ci
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - if: ${{ matrix.os == 'windows-latest' }}
+        run: choco install nasm
+      - if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: sudo apt-get update
+      - if: ${{ matrix.os != 'windows-latest' }}
+        uses: hendrikmuhs/ccache-action@v1.2
```

Comment:
Absolutely :-) I went with full shas, rather than short shas; while I was there it seemed better to be explicit

But more than happy to swap it to short ones if full ones would cause any issues :-)